packages/ai/core/generate-object/generate-object.ts

```diff
@@ -152,59 +152,33 @@ Default and recommended: 'auto' (best mode for the model).
         currentDate?: () => Date;
       };
     },
-): Promise<GenerateObjectResult<TYPE>>;
-
-export async function generateObject<SCHEMA, RESULT>({
-  model,
-  enum: enumValues, // rename bc enum is reserved by typescript
-  schema: inputSchema,
-  schemaName,
-  schemaDescription,
-  output = 'object',
-  system,
-  prompt,
-  messages,
-  maxRetries: maxRetriesArg,
-  abortSignal,
-  headers,
-  experimental_repairText: repairText,
-  experimental_telemetry: telemetry,
-  providerOptions,
-  _internal: {
-    generateId = originalGenerateId,
-    currentDate = () => new Date(),
-  } = {},
-  ...settings
-}: Omit<CallSettings, 'stopSequences'> &
-  Prompt & {
-    /**
-     * The expected structure of the output.
-     *
-     * - 'object': Generate a single object that conforms to the schema.
-     * - 'array': Generate an array of objects that conform to the schema.
-     * - 'no-schema': Generate any JSON object. No schema is specified.
-     *
-     * Default is 'object' if not specified.
-     */
-    output?: 'object' | 'array' | 'enum' | 'no-schema';
-
-    model: LanguageModel;
-    enum?: Array<SCHEMA>;
-    schema?: z.Schema<SCHEMA> | Schema<SCHEMA>;
-    schemaName?: string;
-    schemaDescription?: string;
-    experimental_repairText?: RepairTextFunction;
-    experimental_telemetry?: TelemetrySettings;
-    providerOptions?: ProviderOptions;
-
-    /**
-     * Internal. For test use only. May change without notice.
-     */
-    _internal?: {
-      generateId?: () => string;
-      currentDate?: () => Date;
-    };
-  }): Promise<GenerateObjectResult<RESULT>> {
+): Promise<DefaultGenerateObjectResult<TYPE>> {
```

Comment:
We should return `GenerateObjectResult<TYPE>` (not the default, that's internal)

Reply:
Aha yeah I wasn't sure, will update 🙏 I think I got a type error will see why 👍

Reply:
also can you change `TYPE` to `RESULT` (which is more specific, a generic will always be a type)

Reply:
done 🙏